mlrun/utils/helpers.py

```diff
@@ -660,11 +660,18 @@ def dict_to_json(struct):
 
 
 def parse_artifact_uri(uri, default_project=""):
-    uri_pattern = r"^((?P<project>.*)/)?(?P<key>.*?)(\#(?P<iteration>.*?))?(:(?P<tag>.*?))?(@(?P<uid>.*))?$"
+    """parse artifact uri into project, key, tag, iter, tree
+    uri format: [<project>/]<key>[#<iter>][:<tag>][@<tree>]
+
+    :param uri:            uri to parse
+    :param default_project: default project name if not in uri
+    :returns: project, key, iter, tag, tree
+    """
+    uri_pattern = r"^((?P<project>.*)/)?(?P<key>.*?)(\#(?P<iteration>.*?))?(:(?P<tag>.*?))?(@(?P<tree>.*))?$"
```

Comment:
Not sure I follow why uid is no longer part of the artifact uri

Reply:
The `tree` was called `uid` before the refactor, and this is what actually was in the URI.
Now, UID is auto-generated in the BE, and this value was never a part of the URI.